packages/core/assemblyManager.js

```diff
@@ -4,6 +4,17 @@ import { readConfObject } from './configuration'
 
 export default self => ({
   views: {
+    async getCanonicalRefName(refName, assemblyName) {
+      const refNameAliases = await self.getRefNameAliases(assemblyName)
+      const aliasesToCanonical = {}
+      Object.entries(refNameAliases).forEach(([ref, aliases]) => {
+        aliases.forEach(alias => {
+          aliasesToCanonical[alias] = ref
+        })
+        aliasesToCanonical[ref] = ref
+      })
+      return aliasesToCanonical[refName]
+    },
```

Comment:
Just an idea, not even sure if it would work, but MST caches any view that's accessed with `get`, so if you could create a `get` view you could avoid calling `getRefNameAliases` each time (which might be a bit slow since it involves an RPC call). Maybe a `get canonicalNameData() {}` that figures out the canonical names for all the assemblies in `assemblyData`, and then `getCanonicalRefName` would just use that?

Reply:
We discussed concepts like this and we in general think that there is room for improvement, but I it's hard. The code currently does not need to be high performance since it is applied to a single feature refname (and its mate) at a time, so it can kind of take a hit with the multiple getRefNameAliases, but it could definitely use optimization if it was applied. I think we may need to think about strategies for that. The get canonicalNameData would still invoke an async which would need to be carefully resolved so that potentially it is undefined when unresolved, and defined in some other way when resolved, and then everything else would need to be reactive to that. We also discussed how async stuff can kind of confuse the mobx concept entirely, and how we gotta be pretty careful, but it's not clear if there is any actual problem in the current implementation. I was thinking that I might just merge this code as is for now, and we can think about ways to restructure this in the future

Reply:
Yeah, I'm fine with merging for now. Just was trying to get some ideas out there.